### PR TITLE
roachtest: add roachtest for GORM

### PR DESCRIPTION
**Pull request by RichardJCai**
roachtest: add roachtest for GORM

Release note: None

Resolves #63958 

Not sure if we want to use secure here or not, their test originally is hardcoded to SSLMODE=disable but we can enable.

### CODE REVIEW

File: pkg/cmd/roachtest/gorm_helpers.go [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+		db.Logger = db.Logger.LogMode(logger.Silent)
+	}
+
+	return
+}
+
+func RunMigrations() {
+	var err error
+	allModels := []interface{}{&User{}, &Account{}, &Pet{}, &Company{}, &Toy{}, &Language{}}
+	rand.Seed(time.Now().UnixNano())
```
Comment (RichardJCai) [inline pkg/cmd/roachtest/gorm_helpers.go:116]: This is giving the linter trouble (wants timeutil.Now()), is there a way we can tell the linter to ignore this?

**Review by rafiss — APPROVED** [review]
awesome! lgtm after comments and linting is fixed

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/63958#-:-MYpwo4FC767AQiDc0ZM:bwuoxz5)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @RichardJCai)

___
*[pkg/cmd/roachtest/gorm_helpers.go, line 14 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/63958#-MYpwfZj-EMGcKr-0f27:-MYpwfZj-EMGcKr-0f28:b-96ys5m) ([raw file](https://github.com/cockroachdb/cockroach/blob/8753b70a6a97c4c8db6602f6fc8f7a50c4a0423d/pkg/cmd/roachtest/gorm_helpers.go#L14)):*
> ```Go
> 
> const gormTestHelperGoFile = `package tests_test
> ```

maybe leave a comment somewhere above saying what you had to change
___
*[pkg/cmd/roachtest/gorm_helpers.go, line 116 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/63958#-MYlBoIb9C0mufrQlLud-r1-116:-MYpwXg43JtB0qGZhR5z:b-7mspae) ([raw file](https://github.com/cockroachdb/cockroach/blob/8753b70a6a97c4c8db6602f6fc8f7a50c4a0423d/pkg/cmd/roachtest/gorm_helpers.go#L116)):*
<details><summary><i>Previously, RichardJCai (Richard Cai) wrote…</i></summary><blockquote>

This is giving the linter trouble (wants timeutil.Now()), is there a way we can tell the linter to ignore this?
</blockquote></details>

i think it might be `//nolint:timer`

**Review by RichardJCai — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/63958)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @rafiss)

___
*[pkg/cmd/roachtest/gorm_helpers.go, line 14 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/63958#-MYpwfZj-EMGcKr-0f27:-MYq8qml1OxXcfwKoj3H:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/8753b70a6a97c4c8db6602f6fc8f7a50c4a0423d/pkg/cmd/roachtest/gorm_helpers.go#L14)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

maybe leave a comment somewhere above saying what you had to change
</blockquote></details>

Done.
___
*[pkg/cmd/roachtest/gorm_helpers.go, line 116 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/63958#-MYlBoIb9C0mufrQlLud-r1-116:-MYq8rG68dYU0Wvd2ykV:bvxhijp) ([raw file](https://github.com/cockroachdb/cockroach/blob/8753b70a6a97c4c8db6602f6fc8f7a50c4a0423d/pkg/cmd/roachtest/gorm_helpers.go#L116)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

i think it might be `//nolint:timer`
</blockquote></details>

Trying this, didn't seem to work when I ran `make lintshort`

**Comment by RichardJCai** [comment]
Thanks! 
bors r= rafiss

**Pull request merged**